src/ucs/async/async.c

```diff
@@ -584,13 +586,8 @@ ucs_status_t ucs_async_remove_handler(int id, int is_sync)
 
     async = handler->async;
     if (async != NULL) {
-        ucs_mpmc_queue_lock(&async->missed);
-        ucs_mpmc_queue_for_each(elem, &async->missed) {
-            if (ucs_async_missed_event_unpack_id(elem->value) == handler->id) {
-                elem->value = UCS_ASYNC_INVALID_EVENT;
-            }
-        }
-        ucs_mpmc_queue_unlock(&async->missed);
+        ucs_mpmc_queue_remove_if(&async->missed, ucs_async_missed_event_is_same,
+                                 (void *)(uintptr_t)handler->id);
```

Comment:
can pass a pointer to the handler itself as "arg", it could simplify the code a bit